### PR TITLE
chore: Propagate Ruff failures from complexity checks

### LINKED ISSUE #2377: Propagate Ruff failures from complexity checks
## Summary
The complexity validator reports success when Ruff fails, targets are missing, or output is empty/invalid.

## Evidence
- `validation/complexity.py:40-63,103-107`
- `tests/unit/validation/test_complexity.py:43-46`

## Acceptance criteria
- [ ] Treat nonzero Ruff exits, missing targets, empty output, and invalid JSON as tool failures.
- [ ] Surface stderr and structured JSON errors.
- [ ] Preserve success for a valid zero-finding run.
- [ ] Add regression tests for every failure boundary.

_Reported by a source/test-scoped Athena repository review._

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary
Implements the requested changes for issue #2377.

## Changes
See the PR diff for the full change set.

## Testing
Not run by the automation pipeline.

## Closes
Closes #2377

Generated by Hephaestus automation

### CODE REVIEW

File: hephaestus/validation/complexity.py [hunk 1]
```diff
@@ -147,7 +217,17 @@ def main() -> int:
     repo_root = resolve_repo_root(args)
 
     if args.json:
-        violations = run_ruff_complexity_check(args.path, args.threshold, repo_root)
+        try:
```
Comment (mvillmow) [inline hephaestus/validation/complexity.py:241]: [Review] `subprocess.run(..., timeout=NETWORK_TIMEOUT)` can raise `TimeoutExpired` before returning a result, but this new handler catches only `RuffComplexityError`. A hung Ruff process therefore bypasses both this JSON envelope and normal mode's false return, producing a traceback. Convert `TimeoutExpired` (and process-launch `OSError`) to `RuffComplexityError` at the runner boundary and add timeout coverage for both CLI modes; this preserves the tool-failure contract and POLA.

Comment (mvillmow) [reply hephaestus/validation/complexity.py:241]: [Response] Converted Ruff timeout and process-launch failures into structured RuffComplexityError handling, with regression coverage for normal and JSON CLI modes.
